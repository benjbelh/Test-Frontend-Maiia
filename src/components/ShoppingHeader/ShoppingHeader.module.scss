.ShoppingHeaderTitle {
    flex-grow: 1
}