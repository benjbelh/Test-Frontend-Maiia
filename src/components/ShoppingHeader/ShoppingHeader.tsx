import { AppBar, Badge, Fade, IconButton, Paper, Popper, Toolbar, Typography } from '@material-ui/core';
import { ShoppingBasket } from '@material-ui/icons';
import CartContainer from 'components/CartContainer';
import React, { FunctionComponent } from 'react';
import { useSelector } from 'react-redux';
import { cartItemNumberSelector } from 'store/Cart/selectors';
import styles from './ShoppingHeader.module.scss';

type ShoppingHeaderProps = {
  title?: string;
};

export const ShoppingHeader: FunctionComponent<ShoppingHeaderProps> = ({ title = '' }) => {
  const cartItemNumber = useSelector(cartItemNumberSelector);

  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);

  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorEl(anchorEl ? null : event.currentTarget);
  };

  const open = Boolean(anchorEl);

  return (
    <AppBar position="sticky">
      <Toolbar>
        <Typography variant="h6" className={styles.ShoppingHeaderTitle}>
          {title}
        </Typography>
        <IconButton aria-label="shooping cart" color="inherit" onClick={handleClick}>
          <Badge badgeContent={cartItemNumber} color="secondary">
            <ShoppingBasket />
          </Badge>
        </IconButton>
        <Popper
          placement="bottom-end"
          disablePortal={false}
          modifiers={{
            flip: {
              enabled: true,
            },
            preventOverflow: {
              enabled: true,
              boundariesElement: 'scrollParent',
            },
          }}
          open={open}
          anchorEl={anchorEl}
          transition
        >
          {({ TransitionProps }) => (
            <Fade {...TransitionProps} timeout={350}>
              <Paper>
                <CartContainer />
              </Paper>
            </Fade>
          )}
        </Popper>
      </Toolbar>
    </AppBar>
  );
};

export default ShoppingHeader;
