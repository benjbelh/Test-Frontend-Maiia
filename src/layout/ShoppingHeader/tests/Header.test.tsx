import React from 'react';
import { render } from '@testing-library/react';
import ShoppingHeader from '../ShoppingHeader';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { initialState } from '../../../store/initStore.mock';

describe('ShoppingHeader', () => {
  it('should renders correctly', () => {
    const mockStore = configureStore();
    const store = mockStore(initialState);
    const { asFragment } = render(
      <Provider store={store}>
        <ShoppingHeader title="" />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
