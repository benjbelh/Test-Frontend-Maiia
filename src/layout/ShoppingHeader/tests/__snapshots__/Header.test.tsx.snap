// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingHeader should renders correctly 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        class="MuiTypography-root ShoppingHeaderTitle MuiTypography-h6"
      />
      <button
        aria-label="shopping cart"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
        data-testid="shopping-cart"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <span
            class="MuiBadge-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17.21 9l-4.38-6.56c-.19-.28-.51-.42-.83-.42-.32 0-.64.14-.83.43L6.79 9H2c-.55 0-1 .45-1 1 0 .09.01.18.04.27l2.54 9.27c.23.84 1 1.46 1.92 1.46h13c.92 0 1.69-.62 1.93-1.46l2.54-9.27L23 10c0-.55-.45-1-1-1h-4.79zM9 9l3-4.4L15 9H9zm3 8c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
              />
            </svg>
            <span
              class="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorSecondary MuiBadge-invisible"
            >
              0
            </span>
          </span>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
</DocumentFragment>
`;
